openedx/core/djangoapps/schedules/management/commands/tests/test_send_email_base_command.py

```diff
@@ -41,6 +43,17 @@ def test_weeks_option(self):
             self.command.handle(site_domain_name=self.site.domain, date='2017-09-29', weeks=12)
             self.assertEqual(enqueue.call_count, 12)
 
+    def test_check_completion_option_with_completion_enabled(self):
+        self.override_waffle_switch(True)
+        with patch.object(self.command, 'send_emails') as send_emails:
+            self.command.handle(site_domain_name=self.site.domain, date='2017-09-29', check_completion=True)
+            send_emails.assert_called_once_with(
+                self.site,
+                datetime.datetime(2017, 9, 29, tzinfo=pytz.UTC),
+                None,
+                True
+            )
```

Comment:
This only tests that the management command's `check_completion` command gets passed through, and has nothing to do with the completion waffle flag or block completion

To test that the waffle flag and block completion actually have an effect on the emails that get sent, you'll need to add tests to `openedx/core/djangoapps/schedules/tests/test_resolvers.py`.

Reply:
Added new tests